### PR TITLE
contracts/ens: Add ENS contract binding

### PR DESCRIPTION
**Pull request by Arachnid**

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Fri Aug 26 12:01:25 UTC 2016_

**Comment by fjl** [comment]
```text
contracts\ens\ens_test.go:21: github.com/ethereum/go-ethereum/core.GenesisAccount composite literal uses unkeyed fields
```

**Comment by Arachnid** [comment]
@fjl done

**Comment by fjl** [comment]
Please add LGPL license headers to all files including the generated wrapper. Just copy from any file and make sure the year is 2016.

**Comment by Arachnid** [comment]
PTAL.

File: contracts/ens/ens.go [hunk 1]
```diff
@@ -1,5 +1,21 @@
 //go:generate abigen --sol contract/ENS.sol --pkg contract --out contract/ens.go
```
Comment (fjl) [inline contracts/ens/ens.go]: Please move this line below the package declaration. I should've said that earlier, sorry.

File: contracts/ens/ens_test.go [hunk 2]
```diff
@@ -1,3 +1,18 @@
...
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
 package ens
```
Comment (fjl) [inline contracts/ens/ens_test.go]: There needs to be a blank line between license and the package declaration.

File: contracts/ens/contract/ens.go [hunk 3]
```diff
@@ -0,0 +1,900 @@
+// This file is an automatically generated Go binding. Do not modify as any
+// change will likely be lost upon the next re-generation!
```
Comment (fjl) [inline contracts/ens/contract/ens.go]: Please replace this comment with the license comment.

File: contracts/ens/contract/ens.sol [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
```
Comment (fjl) [inline contracts/ens/contract/ens.sol]: There is no real need to add the header to the contract source code unless you (as the ENS author) choose to provide it under the terms of the LGPLv3. Adding the header will also sign the copyright over to "the go-ethereum Authors". 

Please consider whether you want to keep it that way.

File: contracts/ens/contract/ens.go (hunk 3 above)
Comment (fjl) [inline contracts/ens/contract/ens.go]: OK, we agreed to blacklist the wrapper instead

**Pull request merged**